### PR TITLE
GroupImportFixer - fix breaking code when fixing root classes

### LINKED ISSUE #5574: `group_import`breaks code when fixing root classes
Actual output:

```diff
- use ReflectionClass;
- use ReflectionMethod;
+ use R\{ReflectionClass, ReflectionMethod};
```

Expected output:

```diff
- use ReflectionClass;
- use ReflectionMethod;
+ use \{ReflectionClass, ReflectionMethod};
```

- Command: `php-cs-fixer fix --config=.php_cs.dist --dry-run --diff --verbose --stop-on-violation`
- Version: PHP CS Fixer 2.18.4-DEV Remote Void
- Runtime: PHP 7.4.14

### PR DESCRIPTION
**Pull request by Leprechaunz**
Fixes #5574

### CODE REVIEW

**Comment by kubawerlos** [comment]
@Leprechaunz you don't want to add that test in integration tests, but in `GroupImportFixerTest`.

File: src/Fixer/Import/GroupImportFixer.php [hunk 1]
```diff
@@ -188,7 +188,12 @@ private function addGroupUseStatements(array $statements, Tokens $tokens)
      */
     private function getNamespaceNameWithSlash(NamespaceUseAnalysis $useDeclaration)
     {
-        return substr($useDeclaration->getFullName(), 0, strripos($useDeclaration->getFullName(), '\\') + 1);
+        $position = strripos($useDeclaration->getFullName(), '\\');
```
Comment (kubawerlos) [inline src/Fixer/Import/GroupImportFixer.php:191]:
```suggestion
        $position = strrpos($useDeclaration->getFullName(), '\\');
```
We don't need case insensitive check here.

File: tests/Fixer/Import/GroupImportFixerTest.php [hunk 2]
```diff
@@ -217,6 +217,21 @@ public function provideFixCases()
...
+use Foo\{Bar, Baz};
+use ReflectionClass;
+use ReflectionMethod;
+',
+                '<?php
+
+use Foo\Bar;
+use Foo\Baz;
+use ReflectionClass;
+use ReflectionMethod;
```
Comment (kubawerlos) [inline tests/Fixer/Import/GroupImportFixerTest.php:234]: Let's add one more test when all imported classes have leading `\`.

Comment (Leprechaunz) [reply tests/Fixer/Import/GroupImportFixerTest.php:234]: Good point, thanks

**Comment by Leprechaunz** [comment]
@kubawerlos Could you please link this issue to this PR? https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5574
I think I don't have the right to do this.

**Review by kubawerlos — APPROVED** [review]

**Comment by keradus** [comment]
Thanks for fixing this bug @Leprechaunz.

**Comment by weshooper** [comment]
Thanks @Leprechaunz 😍

**Pull request merged**